accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -537,14 +540,69 @@ separately.
 ### Runtime checks and weak checking
 
 When weak checking is enabled, runtime type tests (including explicit and
-implicit casts) shall succeed with a warning whenever the runtime type test
-would have succeeded if all `?` types were ignored, `Never` were treated as
-`Null`, and `required` named parameters were treated as optional.
-
-In legacy libraries, whether run in strong or weak checking mode, instance
-checks (`e is T`) also return `false` whenever they would have done so pre-nnbd.
-Specifically, if `e` evaluates to `null`, then `e is T` always returns `false`
-unless `T <: Null` or `Object <: T`.
+implicit casts) shall succeed whenever the runtime type test would have
+succeeded if all `?` types were ignored, `Never` were treated as `Null`,
+`Object` were treated as nullable, and `required` named parameters were treated
+as optional.
+
+Instance checks (`e is T`) and casts (`e as T`) behave differently when run in
+strong vs weak checking mode.
+
+Let `LEGACY_SUBTYPE(S, T)` be true iff `S` is a subtype of `T` with all `?`
+types ignored, `Never` treated as `Null`, `Object` treated as nullable, and
+`required` parameters treated as optional.
+
+Let `NNBD_SUBTYPE(S, T)` be true iff `S` is a subtype of `T` as specified in the
+[NNBD subtyping rules](https://github.com/dart-lang/language/blob/master/resources/type-system/subtyping.md).
+
+We define the weak checking and strong checking mode instance tests as follows:
+
+**In weak checking mode**: if `e` has runtime type `S`, an instance check `e is T`
+**whether defined in a legacy or opted-in library** is evaluated as follows:
+  - If `S` is `Null` return `LEGACY_SUBTYPE(T, NULL) || LEGACY_SUBTYPE(Object,
+    T)`
+  - Otherwise return `LEGACY_SUBTYPE(S, T)`
+
+**In strong checking mode**: if `e` has runtime type `S`, an instance check `e is T`
+defined in a **legacy library** is evaluated as follows:
+  - If `S` is `Null` return `NNBD_SUBTYPE(T, NULL) || NNBD_SUBTYPE(Object, T)`
+  - Otherwise return `NNBD_SUBTYPE(S, T)`
+
+**In strong checking mode**: if `e` has runtime type `S`, an instance check `e is T`
+defined in an **opted-in library** is evaluated as follows:
+  - return `NNBD_SUBTYPE(S, T)`
+
+We define the weak checking and strong checking mode casts as follows:
+
+**In weak checking mode**: if `e` has runtime type `S`, a cast `e as T`
+**whether defined in a legacy or opted-in library** is evaluated as follows:
+  - return `LEGACY_SUBTYPE(S, T)`
```

Comment:
Sorry, I was missing the additional requirement that `Null` be treated as `bottom` of the type hierarchy again.  I've re-stated the definition of `LEGACY_SUBTYPE` above.  I believe that it can be viewed either as treating all types as `*` types, or by directly re-axiomatizing `Never` and `Null` as bottom and `Object` as top.

`null as int?` should succeed, as should `null as int`.